include/aws/crt/StringView.h

```diff
@@ -0,0 +1,136 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#if __cplusplus >= 201703L
+#    include <string_view>
+#else
+#    include <algorithm>
+#    include <cassert>
+#    include <iterator>
+#    include <type_traits>
+#endif
+
+namespace Aws
+{
+    namespace Crt
+    {
+#if __cplusplus >= 201703L
+        using StringView = std::string_view;
+#else
+        class StringView
+        {
+          public:
+            typedef char *pointer;
+            typedef const char *const_pointer;
+            typedef char &reference;
+            typedef const char &const_reference;
+            typedef const_pointer const_iterator;
+            typedef const_iterator iterator;
+            typedef std::reverse_iterator<const_iterator> const_reverse_iterator;
+            typedef const_reverse_iterator reverse_iterator;
+            static constexpr const size_t npos = static_cast<size_t>(-1);
+
+            StringView() noexcept : m_size(0), m_data(nullptr) {}
+            StringView(const StringView &other) noexcept = default;
+            StringView(const char *s, size_t size) noexcept : m_size(size), m_data(s) {}
+            StringView(const char *s) noexcept : m_size(std::char_traits<char>::length(s)), m_data(s) {}
```

Comment:
These 2 constructors are NOT noexcept in the [standard](https://en.cppreference.com/w/cpp/string/basic_string_view/basic_string_view) (don't know why though)